mso/datasource_mso_tenant.go

```diff
@@ -18,54 +19,128 @@ func datasourceMSOTenant() *schema.Resource {
 		SchemaVersion: version,
 
 		Schema: (map[string]*schema.Schema{
+
 			"name": &schema.Schema{
 				Type:         schema.TypeString,
 				Required:     true,
 				ValidateFunc: validation.StringLenBetween(1, 1000),
 			},
 
 			"display_name": &schema.Schema{
-				Type:         schema.TypeString,
-				Required:     true,
-				ValidateFunc: validation.StringLenBetween(1, 1000),
+				Type:     schema.TypeString,
+				Computed: true,
 			},
 
 			"description": &schema.Schema{
 				Type:     schema.TypeString,
-				Optional: true,
 				Computed: true,
 			},
 
+			"orchestrator_only": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+
 			"user_associations": &schema.Schema{
 				Type: schema.TypeSet,
 				Elem: &schema.Resource{
 					Schema: map[string]*schema.Schema{
 						"user_id": {
 							Type:     schema.TypeString,
-							Optional: true,
+							Computed: true,
 						},
 					},
 				},
-				Optional: true,
 				Computed: true,
 			},
 
 			"site_associations": &schema.Schema{
-				Type: schema.TypeSet,
```

Comment:
There is no impact other then bugfix. Typeset is only sets one site_association in statefile and TypeList allows for all to be set correctly.

Just retested and confirmed there is no impact with this change for user_associations, since I noticed that I did not change this in the data source. So migrate from typeSet to typeList has no impact in data source behaviour and the existing statefile